src/util/thpool/thpool.c

```diff
@@ -347,19 +350,15 @@ static void *thread_do(struct thread *thread_p) {
 	}
 
 	/* Mark thread as alive (initialized) */
-	pthread_mutex_lock(&thpool_p->thcount_lock);
-	thpool_p->num_threads_alive += 1;
-	pthread_mutex_unlock(&thpool_p->thcount_lock);
+	++thpool_p->num_threads_alive;
 
 	while(threads_keepalive) {
 
 		bsem_wait(thpool_p->jobqueue.has_jobs);
 
 		if(threads_keepalive) {
 
-			pthread_mutex_lock(&thpool_p->thcount_lock);
-			thpool_p->num_threads_working++;
-			pthread_mutex_unlock(&thpool_p->thcount_lock);
+			++thpool_p->num_threads_working;
```

Comment:
why `++thpool_p->num_threads_working;` instead of `thpool_p->num_threads_working++;` ?

Reply:
Are there any writings you can refer me to which discuss this ?